### DISCUSSION TITLE
Connectkit Siwe for Next 14 / app router

### DISCUSSION THREAD

**Discussion by elmariachi111** [family/connectkit#412] [Ideas]
seems as if the server side siwe package only supports pages router / Next's old server API. Also, the siwe package mixes frontend / react code into a server side include, which breaks my Next 14 build with a webpack error.

I spent some time rewriting the server side code and have put it into a separate file. Tried building the connectkit codebase but failed, so I thought I'll provide this as a gist: https://gist.github.com/elmariachi111/dd73eda88b1aacaa7146a2e31cc21024 feel free to use that code to save yourself from api fiddling.

**Comment by lochie** [comment]
This has been on my list for a while now, though still unsure if it should be a new package as `pages` still needs to be supported.

Thanks for the code, it should be good help when working through this feature!

**Reply by elmariachi111** [reply]
Think, we can abstract the request at a level between the routing layer and the handlers. Clients import the layer their framework version needs, the other side is treeshaken. Imo the frontend / provider code also must be separated from the server code and add a use client directive.
